app/merger.go

```diff
@@ -48,14 +48,15 @@ func (smartMerger) Merge(reports []report.Report) report.Report {
 	case 1:
 		return reports[0]
 	}
-	c := make(chan report.Report, l)
-	for _, r := range reports {
-		c <- r
+	c := make(chan *report.Report, l)
+	for i := range reports {
+		c <- &reports[i]
 	}
 	for ; l > 1; l-- {
-		go func(left, right report.Report) {
-			c <- left.Merge(right)
+		go func(left, right *report.Report) {
+			r := left.Merge(*right)
+			c <- &r
 		}(<-c, <-c)
 	}
-	return <-c
+	return *<-c
```

Comment:
i've modified the function once again to use your approach, it's definitely the cleanest solution.